### PR TITLE
Added filter to show only contextual logs from App Insights.

### PR DESCRIPTION
**Pull request by RohitRanjanMS**
All live telemetry coming from an application being slumped together with no ability to differentiate between separate runs (e.g., a regular scheduled Function run vs an execution of the same Function in the "Test + Run" experience on the portal). 

Solution:
Live logs is a feature of Live Diagnostics service which allows differentiation between different execution contexts bases on sessionId. Portal will generate a unique Id for each session, and pass this information in the HTTP headers. Host will include sessionId in all the logs. Logs rendered on the portal will be filtered by sessoinId.

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs [hunk 1]
```diff
@@ -118,6 +119,8 @@ public IActionResult Invoke(string name, [FromBody] FunctionInvocation invocatio
             {
                 { inputParameter.Name, invocation.Input }
             };
+            // LiveLogs sessoin id is used to show only contextual logs in the "Code + Test" experience.
```
Comment (liliankasem) [inline src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs]:
```suggestion
            // LiveLogs session id is used to show only contextual logs in the "Code + Test" experience.
```

File: src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs [hunk 2]
```diff
@@ -42,6 +43,12 @@ public async Task Invoke(HttpContext context)
             var functionExecution = context.Features.Get<IFunctionExecutionFeature>();
             if (functionExecution != null && !context.Response.HasStarted)
             {
+                // LiveLogs sessoin id is used to show only contextual logs in the "Code + Test" experience. The id is included in the custom dimension.
```
Comment (liliankasem) [inline src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs]:
```suggestion
                // LiveLogs session id is used to show only contextual logs in the "Code + Test" experience. The id is included in the custom dimension.
```

**Review by liliankasem — APPROVED** [review]

File: src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs [hunk 3]
```diff
@@ -130,7 +133,19 @@ public IActionResult Invoke(string name, [FromBody] FunctionInvocation invocatio
...
                     {
-                        await scriptHost.CallAsync(function.Name, arguments);
+                        if (!string.IsNullOrWhiteSpace(sessionId))
+                        {
+                            // Current activity is null due to SuppressFlow. Start a new activity and set baggage so that it's included in the custom dimensions.
+                            Activity activity = new Activity($"{nameof(FunctionsController)}.Invoke");
+                            activity?.Start();
+                            activity?.AddBaggage(ScriptConstants.LiveLogsSessionAIKey, sessionId);
+                            await scriptHost.CallAsync(function.Name, arguments);
+                            activity?.Stop();
```
Comment (brettsam) [inline src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs]: should this be in a `try/finally` to ensure we always call stop?

File: src/WebJobs.Script/ScriptConstants.cs [hunk 4]
```diff
@@ -191,5 +191,6 @@ public static class ScriptConstants
         public static readonly ImmutableArray<string> SystemLogCategoryPrefixes = ImmutableArray.Create("Microsoft.Azure.WebJobs.", "Function.", "Worker.", "Host.");
 
         public static readonly string FunctionMetadataDirectTypeKey = "DirectType";
+        public static readonly string LiveLogsSessionAIKey = "#AzFuncLiveLogsSessionId";
```
Comment (brettsam) [inline src/WebJobs.Script/ScriptConstants.cs:194]: this is the header sent by the portal, right?

File: src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs [hunk 5]
```diff
@@ -42,6 +43,12 @@ public async Task Invoke(HttpContext context)
             var functionExecution = context.Features.Get<IFunctionExecutionFeature>();
             if (functionExecution != null && !context.Response.HasStarted)
             {
+                // LiveLogs session id is used to show only contextual logs in the "Code + Test" experience. The id is included in the custom dimension.
+                string sessionId = context.Request?.Headers[ScriptConstants.LiveLogsSessionAIKey];
+                if (!string.IsNullOrWhiteSpace(sessionId) && Activity.Current != null)
```
Comment (brettsam) [inline src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs]: no need to check for `Activity.Current` null here if you're using the `?` later...

File: src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs (hunk 3 above)
Comment (RohitRanjanMS) [reply src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs]: Done.

File: src/WebJobs.Script/ScriptConstants.cs (hunk 4 above)
Comment (RohitRanjanMS) [reply src/WebJobs.Script/ScriptConstants.cs:194]: yes

File: src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs (hunk 5 above)
Comment (RohitRanjanMS) [reply src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs]: Done.

**Review by brettsam — APPROVED** [review]

**Pull request merged**